net/netns/netns_windows.go

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2020 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package netns
+
+import (
+	"encoding/binary"
+	"log"
+	"syscall"
+	"unsafe"
+
+	"github.com/tailscale/winipcfg-go"
+	"golang.org/x/sys/windows"
+)
+
+// control binds c to the Windows interface that holds a default
+// route, and is not the Tailscale WinTun interface.
+func control(network, address string, c syscall.RawConn) error {
+	if4, err := getDefaultInterface(winipcfg.AF_INET)
+	if err != nil {
+		return err
+	}
+	if err := bindSocket4(c, if4); err != nil {
+		return err
+	}
+
+	if6, err := getDefaultInterface(winipcfg.AF_INET6)
+	if err != nil {
+		return err
+	}
+	if err := bindSocket6(c, if6); err != nil {
+		return err
+	}
+
+	log.Printf("WINDOWS DIAL %s %s, v4=%s v6=%s", network, address, if4, if6)
+
+	return nil
+}
+
+// getDefaultInterface returns the index of the interface that has the
+// non-Tailscale default route for the given address family.
+func getDefaultInterface(family winipcfg.AddressFamily) (ifidx uint32, err error) {
+	ifs, err := winipcfg.GetInterfaces()
+	if err != nil {
+		return 0, err
+	}
+	var tsLUIDs []uint64
```

Comment:
might as well use a map here to simplify the lookup below.

both ways allocate, and this isn't hot, so go for clarity